packages/block-library/src/buttons/editor.scss

```
@@ -3,6 +3,17 @@
 	width: auto;
 }
 
+.wp-block-buttons.is-direction-vertical .wp-block.block-editor-block-list__block[data-type="core/button"] {
```

I tested on this branch after rebasing and the `width` and `margin-left` didn't seem to do anything. There's actually a bit of `margin-right` that looks like it needs to be reset. 

Not sure if it's overriding specificity, but I think the selector can also be reduced to:
```suggestion
.wp-block-buttons.is-direction-vertical .wp-block-button {
```

---

Unfortunately, just using `.wp-block-button` does actually cause `display: inline-block` to be prioritized. It does override specificity. Regardless, I've implemented the suggested changes. Thank you ❤️